lib/endpoints/class-wp-rest-posts-terms-controller.php

```diff
@@ -0,0 +1,225 @@
+		if ( ! in_array( $this->taxonomy, $valid_taxonomies ) ) {
+			return new WP_Error( 'rest_post_taxonomy_invalid', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! get_term_by( 'term_taxonomy_id', $term_id, $this->taxonomy ) ) {
+			return new WP_Error( 'rest_term_invalid', __( "Term doesn't exist." ), array( 'status' => 404 ) );
+		}
+
+		$terms = wp_get_object_terms( $post->ID, $this->taxonomy );
+
+		if ( ! in_array( $term_id, wp_list_pluck( $terms, 'term_taxonomy_id' ) ) ) {
+			return new WP_Error( 'rest_post_not_in_term', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$term = $this->terms_controller->prepare_item_for_response( get_term_by( 'term_taxonomy_id', $term_id, $this->taxonomy ), $request );
+
+		$response = rest_ensure_response( $term );
+
+		return $response;
+	}
+
+	/**
+	 * Add a term to a post
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function create_item( $request ) {
+		$post     = get_post( $request['id'] );
+		$term_id  = absint( $request['term_id'] );
+
+		$post_check = $this->posts_controller->get_item( $request );
+		if ( is_wp_error( $post_check ) ) {
+			return $post_check;
+		}
+
+		$valid_taxonomies = get_object_taxonomies( $post->post_type );
+		if ( ! in_array( $this->taxonomy, $valid_taxonomies ) ) {
+			return new WP_Error( 'rest_post_taxonomy_invalid', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! get_term_by( 'term_taxonomy_id', $term_id, $this->taxonomy ) ) {
+			return new WP_Error( 'rest_term_invalid', __( "Term doesn't exist." ), array( 'status' => 404 ) );
+		}
+
+		$tt_ids = wp_set_object_terms( $post->ID, $term_id, $this->taxonomy, true );
+
+		if ( is_wp_error( $tt_ids ) ) {
+			return $tt_ids;
+		}
+
+		$term = $this->terms_controller->prepare_item_for_response( get_term_by( 'term_taxonomy_id', $term_id, $this->taxonomy ), $request );
+
+		$response = rest_ensure_response( $term );
+		$response->set_status( 201 );
+
+		return $term;
+	}
+
+	/**
+	 * Remove a term from a post.
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|null
+	 */
+	public function delete_item( $request ) {
+		$post     = get_post( $request['id'] );
+		$term_id  = absint( $request['term_id'] );
+
+		$post_check = $this->posts_controller->get_item( $request );
+		if ( is_wp_error( $post_check ) ) {
+			return $post_check;
+		}
+
+		$valid_taxonomies = get_object_taxonomies( $post->post_type );
+		if ( ! in_array( $this->taxonomy, $valid_taxonomies ) ) {
+			return new WP_Error( 'rest_post_taxonomy_invalid', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$terms = wp_get_object_terms( $post->ID, $this->taxonomy );
+		$terms = wp_list_filter( $terms, array( 'term_taxonomy_id' => $term_id ), 'NOT' );
+		$tt_ids = wp_set_object_terms( $post->ID, $terms, $this->taxonomy );
```

Comment:
Worth opening a core ticket for a function to remove a term from an object? I think you might lose term order with this #1214